app/controllers/application_controller.rb

```diff
@@ -65,8 +65,10 @@ def render_404
   private
 
   def update_current_team
-    if current_user.current_team_id.blank? &&
-       current_user.teams.count > 0
+    current_team = Team.find_by_id(current_user.current_team_id)
+    if (current_team.nil? || !current_user.is_member_of_team?(current_team)) &&
+       current_user.teams.count.positive?
+
       current_user.update(
         current_team_id: current_user.teams.first.id
```

Comment:
Wha happens when user is not in any team? (just asking :))

Reply:
Well, theoretically this shouldn't happen :).